### PR TITLE
[libc++] Fix naming of value template parameters

### PR DESCRIPTION
**Pull request by philnik777**

### CODE REVIEW

File: libcxx/.clang-tidy [hunk 1]
```diff
@@ -56,8 +56,12 @@ CheckOptions:
     value: _
   - key:   readability-identifier-naming.TemplateParameterIgnoredRegexp
     value: (.*\:auto|expr-type) # This is https://llvm.org/PR56464
-  - key:   readability-identifier-naming.ValueTemplateParameterIgnoredRegexp # TODO: enforce naming of variable parameters
-    value: .*
+  - key:   readability-identifier-naming.ValueTemplateParameterCase
+    value: lower_case
```
Comment (mordante) [inline libcxx/.clang-tidy:60]: I really dislike this. I really like to stay close to the wording in the Standard an there value arguments are upper case too,

File: libcxx/include/__format/format_arg_store.h [hunk 2]
```diff
@@ -228,15 +228,15 @@ _LIBCPP_HIDE_FROM_ABI void __store_basic_format_arg(basic_format_arg<_Context>*
   ([&] { *__data++ = __format::__create_format_arg<_Context>(__args); }(), ...);
 }
 
-template <class _Context, size_t N>
```
Comment (mordante) [inline libcxx/include/__format/format_arg_store.h:231]: As mentioned in #76829 we typically name these `_Np`. I don't see a reason to deviate in this case.

File: libcxx/include/__mdspan/extents.h [hunk 3]
```diff
@@ -137,9 +137,9 @@ struct __maybe_static_array {
   // static mapping of indices to the position in the dynamic values array
   using _DynamicIdxMap = __static_partial_sums<static_cast<size_t>(_Values == _DynTag)...>;
 
-  template <size_t... Indices>
-  _LIBCPP_HIDE_FROM_ABI static constexpr _DynamicValues __zeros(index_sequence<Indices...>) noexcept {
-    return _DynamicValues{((void)Indices, 0)...};
+  template <size_t... __indices>
```
Comment (mordante) [inline libcxx/include/__mdspan/extents.h:140]: What does the wording look like? I expect `Indices`. I think we have several cases of `_indices __indices` in our code base.

**Review by mordante — COMMENTED** [review]
I'm in favour to fix the non-reserved names, but I'm not to fond to change our parameter style to avoid capitalization. Especially since that seems to be the "typical" use in the Standard,

**Comment by philnik777** [comment]
I don't think I change the style here. We don't have a consistent style for NTTPs in the code base currently. I don't care much either way, but your reasoning seems quite flawed here.

**Comment by mordante** [comment]
> I don't think I change the style here. We don't have a consistent style for NTTPs in the code base currently. I don't care much either way, but your reasoning seems quite flawed here.

I checked the `N` case before and `_Np` seems very common, so I'm not sure why my reasoning is flawed.
The changes to the clang-format style would make `_Np` invalid, which I dislike since the Standard uses `N` quite often.

**Comment by ldionne** [comment]
I have to agree with @mordante here, template parameters are always uppercase in libc++. We haven't been making a distinction whether the template parameter denotes a type or a value, and I don't think we should start naming them differently.

**Comment by philnik777** [comment]
> > I don't think I change the style here. We don't have a consistent style for NTTPs in the code base currently. I don't care much either way, but your reasoning seems quite flawed here.
> 
> I checked the `N` case before and `_Np` seems very common, so I'm not sure why my reasoning is flawed. The changes to the clang-format style would make `_Np` invalid, which I dislike since the Standard uses `N` quite often.

This doesn't make `_Np` invalid. It simply allows both forms, otherwise there would be a lot more cases that changed. IMO the reasoning is flawed because it isn't a change in our conventions. We simply never had any strict convention how to name NTTPs. While the `_[A-Z]p` version is more common, it is not the only one used.

> I have to agree with @mordante here, template parameters are always uppercase in libc++. We haven't been making a distinction whether the template parameter denotes a type or a value, [...].

This is simply wrong. We're using both versions in the code base. If this wasn't the case I wouldn't have to allow both.

**Review by ldionne — APPROVED** [review]
I can get behind this if we strive to move the codebase towards `_Np` style arguments. So we can check this in (this patch should use `_Np` style for the non-uglified ones it introduces though), and then we can do another patch to move towards `_Np` everywhere.

LGTM on this patch w/ `_Np` style for the few ones you added.

**Pull request merged**